Math/NumberTheory/Beta.hs

```diff
@@ -0,0 +1,181 @@
+betasEven eps = (1 / 2) : bets
+  where
+    bets :: [a]
+    bets = zipWith3 (\r1 r2 r3 -> r1 + (-r2) + r3) rhs1 rhs2 rhs3
+
+    odds = [1, 3 ..]
+
+    evens = [0, 2 ..]
+
+    -- [1!, 3!, 5!..]
+    factorial1AsInteger :: [Integer]
+    factorial1AsInteger = skipOdds $ tail factorial
+
+    -- [1!, 3!, 5!..]
+    factorial1 :: [a]
+    factorial1 = map fromInteger factorial1AsInteger
+
+    -- [1 / (2^1 * 1!), 1 / (2^3 * 3!), 1 /(2^5 * 5!), 1 / (2^7 * 7!) ..]
+    fracs :: [Rational]
+    fracs = zipWith (\pow fac -> 1 % (pow * fac)) factorial1AsInteger (iterate (4 *) 2)
+
+    -- First term of the right hand side of (12).
+    rhs1 = zipWith3 (\sgn frac lg -> sgn * lg * approximateValue frac) (cycle [1, -1])
+                                                                       (zipWith Exact odds fracs)
+                                                                       (repeat (log 2))
+
+    -- [1 - (1 / (2^2)), 1 - (1 / (2^4)), 1 - (1 / (2^6)), ..]
+    second :: [a]
+    second = tail $ map (1 -) $ iterate (/ 4) 1
+
+    -- [- (1 - (1 / (2^2))) * zeta(3), (1 - (1 / (2^4))) * zeta(5), - (1 - (1 / (2^6))) * zeta(7), ..]
+    zets :: [a]
+    zets = zipWith3 (\sgn twosFrac z -> sgn * twosFrac * z) (cycle [-1, 1])
+                                                            second
+                                                            (drop 2 $ zetasOdd eps)
+
+    -- [pi / (2^1 * 1!), pi^3 / (2^3 * 3!), pi^5 / (2^5 * 5!), ..]
+    pisAndFacs :: [a]
+    pisAndFacs = zipWith (/) (iterate ((pi * pi) *) pi) (zipWith (*) (iterate (4 *) 2) factorial1)
+
+    -- [[], [pisAndFacs !! 0], [pisAndFacs !! 1, pisAndFacs !! 0], [pisAndFacs !! 2, pisAndFacs !! 1, pisAndFacs !! 0]...]
+    pisAndFacs' :: [[a]]
+    pisAndFacs' = scanl (flip (:)) [] pisAndFacs
+
+    -- Second summand of RHS in (12) for k = [1 ..]
+    rhs2 :: [a]
+    rhs2 = zipWith (*) (cycle [-1, 1]) $ map (sum . zipWith (*) zets) pisAndFacs'
+
+    -- [pi^3 / 4, pi^5 / 6, pi^7 / 8 ..]
+    pis :: [a]
+    pis = map approximateValue $ zipWith Exact (tail odds) (map recip (drop 2 evens))
+
+    -- [[3!, 5!, 7! ..], [5!, 7! ..] ..]
+    oddFacs :: [[a]]
+    oddFacs = iterate tail (tail factorial1)
+
+    -- [1, 4, 16 ..]
+    fours :: [a]
+    fours = iterate (4 *) 1
+
+    -- [[3! * 2^0, 5! * 2^2, 7! * 2^4 ..], [5! * 2^0, 7! * 2^2, 9! * 2^4 ..] ..]
+    infSumDenoms :: [[a]]
+    infSumDenoms = map (zipWith (*) fours) oddFacs
+
+    -- [pi^0, pi^2, pi^4, pi^6 ..]
+    pis2 :: [a]
+    pis2 = iterate ((pi * pi) *) 1
+
+    -- [pi^0 * E_1(1), - pi^2 * E_3(1), pi^4 * E_5(1) ..]
+    infSumNum :: [a]
+    infSumNum = zipWith3 (\sgn p eulerP -> sgn * p * eulerP) (cycle [1, -1])
+                                                             pis2
+                                                             (map fromRational . skipOdds . tail $ eulerPolyAt1)
+
+    infSum :: [a]
+    infSum = map (suminf eps . zipWith (/) infSumNum) infSumDenoms
```

Comment:
By the way @Bodigrim, it seems [`suminf`](https://github.com/rockbmb/arithmoi/blob/56d1a848ebee920358304ccfa3b653a70d189ac5/Math/NumberTheory/Zeta.hs#L90) is defined as

```
suminf :: (Floating a, Ord a) => a -> [a] -> a
suminf eps = sum . takeWhile ((>= eps / 111) . abs)
```

Where does the `111` come from?

Reply:
It is a magic number, chosen by trial-and-error method, to satisfy [precision test](https://github.com/cartazio/arithmoi/blob/master/test-suite/Math/NumberTheory/ZetaTests.hs#L92).